transport/nio/src/main/java/org/kaazing/gateway/transport/nio/NioSocketConnector.java

```diff
@@ -97,6 +104,12 @@ protected BridgeServiceFactory initBridgeServiceFactory() {
             ResourceAddress address, IoHandler handler,
             IoSessionInitializer<T> initializer) {
 
+        if (proxyConnector != null) {
+            ConnectFuture future = proxyConnector.connect(address, handler, initializer);
+            if (future != null) {
```

Comment:
I see that you moved the code to create proxy:// to proxy transport component. That seems to be ok but it means the Connector#connect is returning a null ConnectFuture. I don't think any Connector.connect returns null future. We don't have javadoc on the connect methods to explain the contract but my guess is that it returns non-null ConnectFuture object. This is something to think about it.

Other option is to create proxy address here.